src/main/kotlin/com/dto/ReservationManager.kt

```diff
@@ -0,0 +1,61 @@
+package com.dto
+
+import com.excpetion.ReservationAlreadyStartedException
+import com.excpetion.ReservationNotFoundException
+import com.model.ReservationDao
+import com.parkingrate.ParkingRateEvaluator
+import com.util.ReservationCalculator
+import org.springframework.stereotype.Service
+import java.math.BigDecimal
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.Optional
+import javax.transaction.Transactional
+
+
+@Service
+class ReservationManager(private val reservationDao: ReservationDao,
+                         private val reservationConverter: ReservationConverter,
+                         private val reservationCalculator: ReservationCalculator) {
+
+    @Transactional
+    fun createReservation(reservationRequest: ReservationRequest, startTime: LocalDateTime): StartedReservationDto {
+
+        Optional.ofNullable(reservationDao.findStartedReservation(reservationRequest.carLicensePlate))
+                .ifPresent { throw ReservationAlreadyStartedException(reservationRequest.carLicensePlate) }
+
+        return reservationConverter.convertToModel(reservationRequest, startTime)
+                .also { reservationDao.save(it) }
+                .let { reservationConverter.convertToReservationStartedDto(it) }
+    }
+
+    @Transactional
+    fun stopReservation(reservationId: Long, endTime: LocalDateTime): StoppedReservationDto {
```

Comment:
Co się stanie gdy w trakcie wykonywania tej metody (po pobraniu rezerwacji) ktoś inny zmodyfikuje rezerwację w bazie? Czy aktualizacja stawki i daty zakończenia się powiedzie? Czy taki scenariusz jest możliwy?

Reply:
Z tego co wiem to domyślny poziom izolacji zakłada locka na te pole, więc jeśli ktoś spróbuje zaktualizować pole podczas transakcji to jego akcja wykona się po wykonaniu transakcji. Jest oczywiście możliwe aby ktoś bezpośrednio zaktualizował pole w bazie jeśli ma dostęp. Jeśli chodzi od strony api to może wykonać drugą transakcję i pola też się zmienią. Domyślam się że chodzi o implikację aby zablokować możliwość modyfikacji tych pól jeśli update powiódł się za 1 razem.